File: peacecheejecake/leetcode/1_Two_Sum.py
# https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ordered_nums = sorted((k, i) for i, k in enumerate(nums))
        start, end = 0, len(nums) - 1
        while start < end:
            two_sum = ordered_nums[start][0] + ordered_nums[end][0]
            if two_sum < target:
                start += 1
            elif two_sum > target:
                end -= 1
            else:
                return [ordered_nums[start][1], ordered_nums[end][1]]
                

File: sally/leetcode/1_two-sum.cpp
// 1. Two Sum (Easy)
// Runtime: 424 ms, faster than 16.26% of C++ online submissions for Two Sum.
// Memory Usage: 10.1 MB, less than 79.56% of C++ online submissions for Two Sum.

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int N = nums.size();
        bool success = false;
        vector<int> answer;
        for(int i = 0; i < N; i++)
            for(int j = i+1; j < N; j++)
                if(nums[i]+nums[j] == target) {
                    answer.push_back(i);
                    answer.push_back(j);
                }
        return answer;       
    }
};